### PR TITLE
Haz3l delete htyp

### PR DESCRIPTION
**Pull request by disconcision**
Replaces HTyp with Typ

### CODE REVIEW

**Comment by mirryi** [comment]
@disconcision Is this ready? Planning to rebase haz3l-builtins off this.

**Comment by disconcision** [comment]
Probably? Wouldn't mind if @cyrus could take a look at some of the ground stuff in evaluator, not sure I did the right things with unknown types there, but personally I'm fine with merging and sorting it out later

File: src/haz3lcore/dynamics/Evaluator.re [hunk 1]
```diff
@@ -12,31 +12,39 @@ type m('a) = EvaluatorMonad.t('a);
...
+  | NotGroundOrHole(Typ.t) /* the argument is the corresponding ground type */;
 
 type match_result =
   | Matches(Environment.t)
   | DoesNotMatch
   | IndetMatch;
 
-let grounded_Arrow = NotGroundOrHole(Arrow(Hole, Hole));
-let grounded_Sum = NotGroundOrHole(Sum(Hole, Hole));
+//TODO(andrew): should these be unknown internal?
```
Comment (cyrus-) [inline src/haz3lcore/dynamics/Evaluator.re]: lgtm

**Review by cyrus- — APPROVED** [review]

**Comment by cyrus-** [comment]
looks good to me -- go ahead and merge after removing TODO

**Pull request merged**